### PR TITLE
Sci 1326 subjob helpers

### PR DESCRIPTION
**Pull request by msimbirsky**
These are copied vanilla from dx_utils. Let me know if there need to be any changes. Otherwise I will simply add tests.

### CODE REVIEW

**Comment by jdidion** [comment]
Did a major code-cleanup. This module still needs tests.

**Comment by jdidion** [comment]
Tests added. @Damien-Black please review.

File: src/python/test/sugar/__init__.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    with temp_dir() as d:
+        curdir = os.getcwd()
+        os.chdir(d)
+        try:
+            yield d
+        finally:
+            os.chdir(curdir)
+
+
+def run(cmd):
```
Comment (Damien-Black) [inline src/python/test/sugar/__init__.py]: No need to define a top-level `run` command. We already have one:

* [run](https://github.com/dnanexus/dx-toolkit/blob/master/src/python/test/dxpy_testutil.py#L128)

which uses

* [check_output](https://github.com/dnanexus/dx-toolkit/blob/master/src/python/test/dxpy_testutil.py#L71) underneath.

File: src/python/test/sugar/__init__.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import contextlib
+import os
+import shutil
+import subprocess
+import tempfile
+from uuid import uuid4
+
+
+import logging
+logging.basicConfig(level="INFO")
```
Comment (Damien-Black) [inline src/python/test/sugar/__init__.py]: This is fine as is, just a preference thing on my end, change if you want to.

```suggestion
logging.basicConfig(level=logging.INFO)
```

File: src/python/test/sugar/__init__.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+from __future__ import print_function, unicode_literals, division, absolute_import
+import contextlib
+import os
+import shutil
+import subprocess
+import tempfile
```
Comment (Damien-Black) [inline src/python/test/sugar/__init__.py]: I recommend importing our wrapper tempfile [`dxpy_testutil.TemporaryFile`](https://github.com/dnanexus/dx-toolkit/blob/master/src/python/test/dxpy_testutil.py#L651)

File: src/python/dxpy/sugar/chunking.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    return [sizes[i] for i in ids]
+
+
+def as_dxfile_id(dxfile):
+    if isinstance(dxfile, dxpy.DXFile):
+        return dxfile.get_id()
+    elif dxpy.is_dxlink(dxfile):
+        return dxpy.get_dxlink_ids(dxfile)
+    else:
+        return dxfile
```
[the remark this answers is not in the document]
Comment (jdidion) [reply src/python/dxpy/sugar/chunking.py]: The arguments are also allowed to be IDs. I updated the argument names and docstrings.

File: src/python/dxpy/sugar/chunking.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+
+import pysnooper
+@pysnooper.snoop()
+def schedule_lpt(jobs, num_bins):
```
[the remark this answers is not in the document]
Comment (jdidion) [reply src/python/dxpy/sugar/chunking.py]: Agreed. It is required by divide_dxfiles_into_chunks so I made it a function internal to the module.

File: src/python/test/sugar/__init__.py (hunk 3 above)
Comment (jdidion) [reply src/python/test/sugar/__init__.py]: That only works for files, not directories. But I agree that these utility functions should at some point be pushed up into dxpy_testutils.py.

**Comment by jdidion** [comment]
@Damien-Black made suggested changes except where noted. In this and other PRs I have commented out the imports in dxpy.sugar.__init__.py. I will resolve these after we merge all the branches.

**Comment by Damien-Black** [comment]
@jdidion approving for now. Let me know when the conflicts are resolved and I can re-review

**Pull request merged**